### ISSUE TITLE
After CtClass.detach is called, other methods can execute normal.

### ISSUE THREAD

**Issue by Byron4j** [jboss-javassist/javassist#255]
My program like this(dependent javassist version is 3.24.0-GA):

```java
public class Javassist2ClassPool {
    public static void main(String[] args) throws NotFoundException {
        ClassPool classPool = ClassPool.getDefault();
        CtClass cc = classPool.get("org.byron4j.cookbook.javaagent.Javassist2ClassPool");

       
        cc.detach();

        System.out.println(cc.getFields().length); // executed normal.

    }
}
```

After ```CtClass.detach``` is called, other methods can execute normal.
But doc of ```CtClass.detach``` show this:  **```Removes this CtClass object from the ClassPool. After this method is called, any method cannot be called on the removed CtClass object.```**.
